bindings/postmark/postmark.go

```diff
@@ -0,0 +1,154 @@
+	pMeta := postmarkMetadata{}
+
+	// Required properties
+	if val, ok := meta.Properties["serverToken"]; ok && val != "" {
+		pMeta.ServerToken = val
+	} else {
+		return pMeta, errors.New("Postmark binding error: serverToken field is required in metadata")
+	}
+	if val, ok := meta.Properties["accountToken"]; ok && val != "" {
+		pMeta.AccountToken = val
+	} else {
+		return pMeta, errors.New("Postmark binding error: accountToken field is required in metadata")
+	}
+
+	// Optional properties, these can be set on a per request basis
+	pMeta.EmailTo = meta.Properties["emailTo"]
+	pMeta.EmailFrom = meta.Properties["emailFrom"]
+	pMeta.Subject = meta.Properties["subject"]
+	pMeta.EmailCc = meta.Properties["emailCc"]
+	pMeta.EmailBcc = meta.Properties["emailBcc"]
+
+	return pMeta, nil
+}
+
+// Init does metadata parsing and not much else :)
+func (p *Postmark) Init(metadata bindings.Metadata) error {
+	// Parse input metadata
+	meta, err := p.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+
+	// Um, yeah that's about it!
+	p.metadata = meta
+
+	return nil
+}
+
+func (p *Postmark) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{bindings.CreateOperation}
+}
+
+// Write does the work of sending message to Postmark API
+func (p *Postmark) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	// We allow two possible sources of the properties we need,
+	// the component metadata or request metadata, request takes priority if present
+
+	var email postmark.Email
+
+	// Build email from address, this is required
+	if p.metadata.EmailFrom != "" {
+		email.From = p.metadata.EmailFrom
+	}
+	if req.Metadata["emailFrom"] != "" {
+		email.From = req.Metadata["emailFrom"]
+	}
+	if len(email.From) == 0 {
+		return nil, fmt.Errorf("error Postmark from email not supplied")
+	}
+
+	// Build email to address, this is required
+	if p.metadata.EmailTo != "" {
+		email.To = p.metadata.EmailTo
+	}
+	if req.Metadata["emailTo"] != "" {
+		email.To = req.Metadata["emailTo"]
+	}
+	if len(email.To) == 0 {
+		return nil, fmt.Errorf("error Postmark to email not supplied")
+	}
+
+	// Build email subject, this is required
+	if p.metadata.Subject != "" {
+		email.Subject = p.metadata.Subject
+	}
+	if req.Metadata["subject"] != "" {
+		email.Subject = req.Metadata["subject"]
+	}
+	if len(email.Subject) == 0 {
+		return nil, fmt.Errorf("error Postmark subject not supplied")
+	}
+
+	// Build email cc address, this is optional
+	if p.metadata.EmailCc != "" {
+		email.Cc = p.metadata.EmailCc
+	}
+	if req.Metadata["emailCc"] != "" {
+		email.Cc = req.Metadata["emailCc"]
+	}
+
+	// Build email bcc address, this is optional
+	if p.metadata.EmailBcc != "" {
+		email.Bcc = p.metadata.EmailBcc
+	}
+	if req.Metadata["emailBcc"] != "" {
+		email.Bcc = req.Metadata["emailBcc"]
+	}
+
+	// Email body is held in req.Data, after we tidy it up a bit
+	emailBody, _ := strconv.Unquote(string(req.Data))
```

Comment:
Any reason `Unquote` is needed here?

Reply:
I guess to get the original HTML if it contains quotes like attributes in tags, that would be escaped in the json payload.